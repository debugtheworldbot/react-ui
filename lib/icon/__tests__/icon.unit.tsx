import  React from 'react';
import  renderer from 'react-test-renderer'
import Icon from "../icon";
import {mount} from 'enzyme';


describe('icon',()=>{
    it('render test?',()=>{
        const json=renderer.create(<Icon name={'alipay'}/>).toJSON()
        expect(json).toMatchSnapshot()
    })
    it('should response to click event', () => {
        const fn = jest.fn()
        const c = mount(<Icon name={'alipay'} onClick={fn} />)
        c.find('svg').simulate('click')
        expect(fn).toBeCalled()
    })
})