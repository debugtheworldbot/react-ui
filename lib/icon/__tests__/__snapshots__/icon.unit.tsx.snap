// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`icon render test? 1`] = `
<svg
  className="czUi-icon"
>
  <use
    xlinkHref="#alipay"
  />
</svg>
`;
